File: shared/src/commonMain/kotlin/dto/InsuranceProductCategory.kt
package dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class InsuranceProductCategory(
    @SerialName("category_id")
    val id: Int,
    @SerialName("title")
    val title: String,
    @SerialName("product_list")
    val productList: List<InsuranceProduct>,
    @SerialName("show_in_filters")
    val showInFilters: Boolean
)

@Serializable
data class InsuranceProductCategoryList(
    @SerialName("insurance_product_list")
    val products: List<InsuranceProductCategory>
)

File: iosApp/AlfaStrah/Models/Insurances/InsuranceProductCategory.swift
//
//  InsuranceProductCategory.swift
//  AlfaStrah
//
//  Created by Makson on 25.10.2023.
//  Copyright © 2023 Touch Instinct. All rights reserved.
//

// sourcery: transformer
struct InsuranceProductCategory {
    // sourcery: transformer.name = "category_id"
    let id: Int64
    // sourcery: transformer.name = "title"
    let title: String
    // sourcery: transformer.name = "product_list"
    let productList: [InsuranceProduct]
    // sourcery: transformer.name = "show_in_filters"
    let showInFilters: Bool
}
